service/openapi.yaml

```diff
@@ -421,3 +434,17 @@ components:
           examples:
             - 1000
           description: number (in millisecond) to wait before invalidating the cache. If we have cacheInvalidation enabled, the cache can also be evicted if a configuration change happen.
+    metadata: 
+      type: object
+      additionalProperties:
+        oneOf:
+          - type: boolean
+          - type: string
+          - type: number
+    flagMetadataExamples:
+      examples:
+        - team: ecommerce
+          businessPurpose: experiment 
+    flagMetadataDescription:
+      description: Arbitrary metadata for the flag, useful for telemetry and documentary purposes.
```

Comment:
Couldn't you simplify this by doing something like:

```yaml
flagMetadata:
  $ref: "#/components/schemas/metadata"
  description: Arbitrary metadata for the flag, useful for telemetry and documentary purposes.
  examples:
        - team: ecommerce
          businessPurpose: experiment 
```

Then you could get rid of the `allOf` and just reference `#flagMetadata`

Reply:
Never mind, this doesn't seem to be valid in OpenAPI.